index.bs

```diff
@@ -2187,35 +3597,134 @@ dictionary MLLstmOptions {
+        - *options*: an optional {{MLLstmOptions}}. The optional parameters of the operation.
 
-    <div class="note">
+    **Returns:** a sequence of {{MLOperand}}. The first element of the sequence is a 3-D tensor of shape [num_directions, batch_size, hidden_size], the output hidden state from the last time step of the network. The second element is a 3-D tensor of shape [num_directions, batch_size, hidden_size], the output cell state from the last time step of the network. Additionally, if |options|.{{MLLstmOptions/returnSequence}} is set to true, the third element is the 4-D output tensor of shape [steps, num_directions, batch_size, hidden_size] containing every output from each time step in the temporal sequence.
+</div>
+
+<details open>
+  <summary>
+    The {{MLGraphBuilder/lstm(input, weight, recurrentWeight, steps, hiddenSize, options)}} steps are:
+  </summary>
+  <div algorithm=lstm class=algorithm-steps>
+    1. If |options|.{{MLLstmOptions/direction}} is not one of {{MLRecurrentNetworkDirection}}, then [=exception/throw=] a {{TypeError}}.
+    1. Let |num_directions| be `1` if |options|.{{MLLstmOptions/direction}} is `"forward"`, or otherwise let it be `2`.
+    1. [=Assert=]: the type of |input|, |weight| and |recurrentWeight| is {{MLOperand}}.
+        <div class="note">
+            The shape of |input|, |weight| or |recurrentWeight| could be also checked here.
+        </div>
+    1. If |input|.{{MLOperand/[[descriptor]]}}.{{MLOperandDescriptor/dimensions}}[0] is not equal to |steps|, then [=exception/throw=] a "{{DataError}}" {{DOMException}}.
+    1. Let |batch_size| be |input|.{{MLOperand/[[descriptor]]}}.{{MLOperandDescriptor/dimensions}}[1].
+    1. If |options|.{{MLLstmOptions/bias}} [=map/exists=]:
+        1. [=Assert=]: its type is {{MLOperand}}.
+        1. If its rank is not `2`, then [=exception/throw=] a "{{DataError}}" {{DOMException}}.
+        1. If |options|.{{MLLstmOptions/bias}}.{{MLOperand/[[descriptor]]}}.{{MLOperandDescriptor/dimensions}}[0] is not |num_directions|, then [=exception/throw=] a "{{DataError}}" {{DOMException}}.
+        1. If |options|.{{MLLstmOptions/bias}}.{{MLOperand/[[descriptor]]}}.{{MLOperandDescriptor/dimensions}}[1] is not 4 * |hiddenSize|, then [=exception/throw=] a "{{DataError}}" {{DOMException}}.
+    1. If |options|.{{MLLstmOptions/recurrentBias}} [=map/exists=]:
+        1. [=Assert=]: its type is {{MLOperand}}.
+        1. If its rank is not `2`, then [=exception/throw=] a "{{DataError}}" {{DOMException}}.
+        1. If |options|.{{MLLstmOptions/recurrentBias}}.{{MLOperand/[[descriptor]]}}.{{MLOperandDescriptor/dimensions}}[0] is not |num_directions|, then [=exception/throw=] a "{{DataError}}" {{DOMException}}.
+        1. If |options|.{{MLLstmOptions/recurrentBias}}.{{MLOperand/[[descriptor]]}}.{{MLOperandDescriptor/dimensions}}[1] is not 4 * |hiddenSize|, then [=exception/throw=] a "{{DataError}}" {{DOMException}}.
```

Comment:
Please use `4 * |hiddenSize|` instead of `4 * |hiddenSize|` to align with above explanation of **recurrentBias**, and please make same modification for others, thanks.

Reply:
Resolved by moving all names to camelCase.